examples/automatic_speech_recognition/automatic_speech_recognition/evaluator.py

```diff
@@ -0,0 +1,319 @@
+    y_metrics: List[Union[TestSampleMetric, Inference]],
+    binning_info: Optional[Tuple[float, float, float]] = None,  # start, end, num
+    x_logarithmic: bool = False,
+    y_logarithmic: bool = False,
+    metadata: bool = False,
+) -> CurvePlot:
+    y_values = [getattr(m, y_metric) for m in y_metrics]
+    if metadata:
+        x_values = [m.metadata[x_metric] for m in x_metrics]
+    else:
+        x_values = [getattr(m, x_metric) for m in x_metrics]
+
+    if x_logarithmic:
+        bins = list(np.logspace(*binning_info, base=2))
+    else:
+        bins = list(np.linspace(*binning_info))
+
+    bins_centers: List[float] = []
+    bins_values: List[float] = []
+    for i in range(len(bins) - 1):
+        lo, hi = bins[i : i + 2]
+        bin_values = [y for y, x in zip(y_values, x_values) if lo <= x < hi]
+        if len(bin_values) > 0:
+            bins_centers.append(lo + ((hi - lo) / 2))
+            bins_values.append(np.mean(bin_values))
+
+    return CurvePlot(
+        title=f"{y_metric} vs. {x_metric}",
+        x_label=f"{x_metric}",
+        y_label=f"{y_metric}",
+        curves=[Curve(x=bins_centers, y=bins_values)],
+        x_config=AxisConfig(type="log") if x_logarithmic else None,
+        y_config=AxisConfig(type="log") if y_logarithmic else None,
+    )
+
+
+def compute_test_case_plots(
+    complete_metrics: List[TestSampleMetric],
+    test_case_metrics: List[TestSampleMetric],
+    test_samples: List[TestSample],
+) -> List[Plot]:
+    max_word_error = max([metric.WordErrors for metric in complete_metrics])
+
+    return [
+        compute_score_distribution_plot(
+            "WordErrors",
+            test_case_metrics,
+            (0, max_word_error + 1, int(max_word_error)),
+            logarithmic_y=True,
+        ),
+        compute_score_distribution_plot("WordErrorRate", test_case_metrics, (0, 1, 51), logarithmic_y=True),
+        compute_score_distribution_plot("CharacterErrorRate", test_case_metrics, (0, 1, 51), logarithmic_y=True),
+        compute_score_distribution_plot("WordInformationLost", test_case_metrics, (0, 1, 51), logarithmic_y=True),
+        compute_metric_vs_metric_plot(
+            "duration_seconds",
+            "WordErrorRate",
+            test_samples,
+            test_case_metrics,
+            (0, 35, 7),
+            metadata=True,
+        ),
+        compute_metric_vs_metric_plot(
+            "duration_seconds",
+            "CharacterErrorRate",
+            test_samples,
+            test_case_metrics,
+            (0, 35, 7),
+            metadata=True,
+        ),
+        compute_metric_vs_metric_plot(
+            "tempo",
+            "WordErrorRate",
+            test_samples,
+            test_case_metrics,
+            (0, 6, 14),
+            metadata=True,
+        ),
+        compute_metric_vs_metric_plot(
+            "tempo",
+            "CharacterErrorRate",
+            test_samples,
+            test_case_metrics,
+            (0, 6, 14),
```

Comment:
Doesn't need to be fixed immediately, but ideally we wouldn't have hardcoded extents in this file.